src/index.ts

```diff
@@ -568,6 +571,21 @@ export class Traversal<S, A> {
   }
 }
 
+export class At<S, I, A> {
+  readonly _tag: 'At' = 'At'
+  constructor(readonly at: (i: I) => Lens<S, A>) {}
+}
+
+export function atStrMap<A = never>(): At<SM.StrMap<A>, string, Option<A>> {
+  return new At(at => new Lens(s => SM.lookup(at, s), a => s => a.fold(SM.remove(at, s), x => SM.insert(at, x, s))))
+}
+
+export function atSet<A = never>(setoid: Setoid<A>): At<Set<A>, A, boolean> {
```

Comment:
I think that `S.member(setoid)`, `S.insert(setoid)` and `S.remove(setoid)` can be memoized

```ts
export function atSet<A = never>(setoid: Setoid<A>): At<Set<A>, A, boolean> {
  const member = S.member(setoid)
  const insert = S.insert(setoid)
  const remove = S.remove(setoid)
  return new At(
    at => new Lens(s => member(s)(at), a => s => (a ? insert(at, s) : remove(at, s)))
  )
}
```